lib/foreman_ansible_core/roles_reader.rb

```diff
@@ -0,0 +1,36 @@
+module ForemanAnsibleCore
+  class RolesReader
+    class << self
+      def list_roles
+        logger.info('[foreman_ansible] - Reading roles from '\
+                    '/etc/ansible/ansible.cfg roles_path')
+        ansible_config = '/etc/ansible/ansible.cfg'
+        Dir.glob("#{roles_path(ansible_config)}/*").map do |path|
+          path.split('/').last
+        end
+      rescue Errno::ENOENT, Errno::EACCES => e
+        logger.debug("[foreman_ansible] - #{e.backtrace}")
+        raise "[foreman_ansible] - Could not read Ansible config file "\
+          "#{ansible_config} - #{e.message}"
+      end
+
+      def roles_path(ansible_config_file)
+        default_path = '/etc/ansible/roles'
+        roles_line = File.readlines(ansible_config_file).select do |line|
+          line =~ /roles_path/
+        end
+        # Default to /etc/ansible/roles if none found
+        return default_path if roles_line.empty?
+        roles_path_key = roles_line.first.split('=').first.strip
+        # In case of commented roles_path key "#roles_path", return default
+        return default_path unless roles_path_key == 'roles_path'
+        # In case roles_path is there, and is not commented, return the value
+        roles_line.first.split('=').last.strip
+      end
+
+      def logger
+        Foreman::Logging.logger('foreman_ansible')
```

Comment:
This does NOT log in the proxy properly, only in Foreman - which needs to be fixed.